### ISSUE TITLE
Panics are not accessible in Error hooks

### ISSUE THREAD

**Issue by marioizquierdo** [twitchtv/twirp#163]
It is nice that the error hooks are called when there's a panic in a service method implementation. However, the cause of the panic is not available.

### Example

We have a production Twirp service configured with an error hook that sends errors to our error tracker. Usual errors returned by service method implementations are wrapped with `twirp.InternalErrorWith(err)`, which allows our error hook to unwrap the cause like this:

```go
func ErrorHook(ctx context.Context, twerr twirp.Error) context.Context {
	code := twerr.Code()

	// Non-twirp errors are wrapped with twirp.InternalErrorWith(err); try to unwrap
	// that error if possible, which in most cases has a StackTrace.
	err := errorutil.Unwrap(twerr) // could also be pkg/errors.Cause or similar

	reportError(ctx, code, err)
	return ctx
}
```

This is nice because if our implementation returns errors with stacktraces (using something like `pkg/errors.Wrap`) then we can see the stacktraces in our error tracking software.

But when the error is caused by a panic, all we can see in our error tracker is:
```go
code: "internal", error: "Internal service panic"
```

We can not see the panic stacktrace or even the panic description. No clue where that panic came from.

### - Edit - Fixed in #160 

But in case you are in an older version of Twitch, this is the workaround I used to do:

Add HTTP middleware that intercepts the panic and reports that. Which also fixes the issue #162

Doing that results in two errors being reported: once in the Error hook, and then again in the http middleware. You could check for `twerr.Msg() == "Internal service panic"` or similar, but we don't do that because having panics reported twice in our case is fine.

**Comment by spenczar** [comment]
Yeah, we should be able to do better than this. I agree that it's a problem, and I like your proposed solution.

**Comment by spenczar** [comment]
Fixed in #160.
